extensions/objects/templates/SetOfHtmlStringEditor.js

```diff
@@ -42,16 +42,19 @@ oppia.directive('setOfHtmlStringEditor', [
       }
 
       $scope.choices = $scope.initArgs.choices;
-      $scope.choiceGroup = $scope.$parent.value;
+      $scope.selections = $scope.choices.map(function(choice) {
+        return $scope.$parent.value.indexOf(choice.id) !== -1;
+      });
 
       // The following function is necessary to insert elements into the answer
       // groups for the Item Selection Widget.
-      $scope.toggleSelection = function(choice) {
-        var index = $scope.$parent.value.indexOf(choice.id);
-        if (index > -1) {
-          $scope.$parent.value.splice(index, 1);
+      $scope.toggleSelection = function(index) {
+        var choiceHtml = $scope.choices[index].id;
+        var searchIndex = $scope.$parent.value.indexOf(choiceHtml);
```

Comment:
I had some difficulty here. There are two indexes: the one that's passed in (which is the index into $scope.choices) and this one (which is the index into $scope.parent.value). Both represent different things: the former represents all choices, and the latter represents only those which are selected. The problem is that choiceIndex can refer to both.

Having said that, searchIndex isn't a good name either. Maybe selectedChoicesIndex?